python-malformed-yaml/main.py

```diff
@@ -0,0 +1,99 @@
+import os
+import logging
+import sys
+import re
+import yaml
+from github_service import GitHubService as github_service
+
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    level="INFO",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_github_env() -> tuple[str, str, str]:
+    """
+    Function to collect the three required GitHub
+    environmnet variables
+    """
+    token = os.getenv("GITHUB_TOKEN")
+    pr_number = os.getenv("PR_NUMBER")
+    repo = os.getenv("GITHUB_REPOSITORY")
+    if not token:
+        raise ValueError("No GITHUB_TOKEN.")
+    if not pr_number:
+        raise ValueError("No PR_NUMBER.")
+    return token, repo, pr_number
+
+
+def get_changed_yaml_files_from_pr() -> list[str]:
+    """
+    Collect a list of all the new or modified YAML files
+    in a PR, except those in a 'secret/' directory.
+    """
+    token, repository_name, pr = get_github_env()
+    github = github_service(token, repository_name, int(pr))
+    # We assume there must always be some changed or new files in a PR
+    changed_files = github.get_changed_files_from_pr()
+    pattern = re.compile("\\.yml$|\\.yaml$")
+    skip_pattern = re.compile("secret/")
+    changed_yaml_files = [
+        file for file in changed_files if pattern.search(file) and not skip_pattern.search(file)
+    ]
+    if changed_yaml_files:
+        return changed_yaml_files
+    logger.info("🫧 No new or modified YAML files to check.")
+    sys.exit(0)
```

Comment:
So here it would be:
```suggestion
    return [
        file for file in changed_files if pattern.search(file) and not skip_pattern.search(file)
    ]
```

Reply:
Yeah OK I see what you mean

Reply:
Will make changes